django/contrib/admindocs/utils.py

```diff
@@ -246,3 +248,32 @@ def remove_non_capturing_groups(pattern):
 
 def strip_p_tags(value):
     return mark_safe(value.replace("<p>", "").replace("</p>", ""))
+
+
+def get_built_in_tags():
+    tags = []
+    try:
+        engine = Engine.get_default()
+    except ImproperlyConfigured:
+        # Non-trivial TEMPLATES settings aren't supported (#24125).
+        pass
+    else:
+        builtin_libs = [lib for lib in engine.template_builtins]
+        for library in builtin_libs:
+            for tag_name, _ in library.tags.items():
+                tags.append(tag_name)
+    return tags
+
+
+def update_tag_href(text):
+    built_in_tags = get_built_in_tags()
+    # Pattern to match each href and capture the tag part after #
+    pattern = r'href="[a-z/]*\/tags\/#([a-z_]+)"'
+    # Find all hrefs that match the pattern in the text
+    matches = re.findall(pattern, text)
+
+    for tag in matches:
+        if tag in built_in_tags:
+            text = text.replace(f"#{tag}", f"#built_in-{tag}")
+
+    return text
```

Comment:
I think instead of inserting `built_in` into the links, what we want to do is support the `:tag:title <target>` format

We should be able to do that by using a sphinx helper `split_explicit_title` (see https://github.com/sphinx-doc/sphinx/blob/1266c421be03dc9a5cbcf69a7c292eb71d21740d/sphinx/util/nodes.py#L389)
As sphinx is not installed, we should credit it in the docstring if we copy it over

```diff
@@ -101,6 +99,14 @@ ROLES = {
     "tag": "%s/tags/#%s",
 }
 
+def split_explicit_title(text):
+    """Split role content into title and target, if given."""
+    explicit_title_re = re.compile(r'^(.+?)\s*(?<!\x00)<(.*?)>$', re.DOTALL)
+    match = explicit_title_re.match(text)
+    if match:
+        return True, match.group(1), match.group(2)
+    return False, text, text
+
 
 def create_reference_role(rolename, urlbase):
     # Views and template names are case-sensitive.
@@ -109,14 +115,15 @@ def create_reference_role(rolename, urlbase):
     def _role(name, rawtext, text, lineno, inliner, options=None, content=None):
         if options is None:
             options = {}
+        _, title, target = split_explicit_title(text)
         node = docutils.nodes.reference(
             rawtext,
-            text,
+            title,
             refuri=(
                 urlbase
                 % (
                     inliner.document.settings.link_base,
-                    text if is_case_sensitive else text.lower(),
+                    target if is_case_sensitive else target.lower(),
                 )
             ),
             **options,
```